src/operator/controllers/intents_reconcilers/iam/iampolicyagents/azurepolicyagent/agent.go

```diff
@@ -301,3 +323,61 @@ func (a *Agent) vaultAccessPolicyEntryFromIntent(userAssignedIdentity armmsi.Ide
 		},
 	}
 }
+
+func (a *Agent) ensureCustomRolesForIntents(ctx context.Context, userAssignedIdentity armmsi.Identity, intents []otterizev2alpha1.Target) error {
+	// Lock the agent to ensure that no other goroutine is modifying the custom roles in parallel
+	a.roleMutex.Lock()
+	defer a.roleMutex.Unlock()
```

Comment:
Is it enough to lock around ensureCustomRoleForIntent (single) instead? So that the lock is held for a shorter time.

Reply:
moved